addons/test_mail/tests/test_performance.py

```diff
@@ -347,7 +347,7 @@ def test_mail_composer(self):
                 'partner_ids': [(4, customer_id)],
             })
 
-        with self.assertQueryCount(__system__=26, employee=32):
+        with self.assertQueryCount(__system__=30, employee=36):
```

Comment:
Mmmh, quite a lot of extra queries ...

Reply:
In the new version, we only have 3 added queries:
- 2 to notify the record responsible by inbox (get the followers, to know if we display an unfollow link or not for the message in the inbox)
- 1 to send the mail to the customer (also get the followers, to know if we add the unfollow link in the email)
Those queries are in methods that support batch record so if used in batch no additional queries are added but there are often not used in batch mode.

Details: 
Record responsible is notified by inbox so when sending the mail we have 2 additional queries:
- one to get the follower ids of the related record for the message to be notified by inbox
- one to retrieve information about the followers to build a map record ref → follower id
We have chosen a search followed by a read for code readability but a search_read shouldn’t reduce by one the query as its implementation performs a search and then a read.
I think it could be reduced to one query for singleton record, if we implement a special version in that case (only need the follower id, not a map record_ref → id). What do you think ?

Customer is notified by email:
- the added query is to get the follower of the document.
Note that in other similar tests, there may be more queries because:
- for each mail, we verify if the partner has an user which is internal: any(user._is_internal() for user in partner.user_ids)): 2 queries
- for getting the notify link: _notify_get_action_link → _notify_encode_link → 1 query in config_parameter table for data_base_secret: but it should be cached after first call.

In the rating test, the 2 queries to get the followers when notifying by inbox is repeated 100 times as we “apply_rating” 100 times → 200 added queries, each time on one record. But it’s actually 100 users rating each one a record.